node/src/daser.rs

```diff
@@ -0,0 +1,294 @@
+//! Component responsible for data availability sampling of the already synchronized block
+//! headers announced in the Celestia network.
+//!
+//! When a new header is insert in the [`Store`], [`Daser`] gets informed, then it fetches
+//! random [`Sample`]s of the block via Shwap protocol and verifies them. If all samples
+//! get verified successfuly, then block is marked as accepted.
+//!
+//! [`Sample`]: celestia_types::sample::Sample
+
+use std::collections::HashSet;
+use std::sync::Arc;
+
+use celestia_types::ExtendedHeader;
+use cid::Cid;
+use futures::stream::FuturesUnordered;
+use futures::StreamExt;
+use rand::Rng;
+use tokio::select;
+use tokio_util::sync::CancellationToken;
+use tracing::error;
+
+use crate::executor::spawn;
+use crate::p2p::shwap::convert_cid;
+use crate::p2p::{P2p, P2pError};
+use crate::store::{Store, StoreError};
+
+const DEFAULT_MAX_SAMPLES_NEEDED: usize = 16;
```

Comment:
```suggestion
const MAX_SAMPLES_NEEDED: usize = 16;
```

As is, I don't think it's configurable, so we could drop the prefix?

Reply:
Done